### ISSUE TITLE
Static builds stops at `generating static routes` without error when using optimized images

### ISSUE THREAD

**Issue by BryceRussell** [withastro/astro#8547]
### Astro Info

```block
(stackblitz minimal reproduction)
Astro                    v3.0.8
Node                     v16.20.0
System                   Linux (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             none
```

### Describe the Bug

In some cases, when using optimized images, `astro build` will stop/fail at the `generating static route` stage and not throw any errors making production builds appear as if they were successful

**Cases**:
1) [Support Thread](https://discord.com/channels/830184174198718474/1148654506528231434/1148654506528231434) | [Example Repo](https://github.com/scottaw66/sierraalfawhiskey/blob/brokentest/src/content/posts/1.md?plain=1#L11)
    - Breaks when using an inline relative image path in markdown
1) [Support Thread](https://discord.com/channels/830184174198718474/1146556928856707133/1146556928856707133) | [Example](https://discord.com/channels/830184174198718474/1146556928856707133/1146781753516765307) | [Minimal Reproduction](https://github.com/BryceRussell/minimal-repo-md-imgs-in-public)
    - Breaks when using an inline relative image path in markdown to asset in `/public`
1) [Support Thread](https://discord.com/channels/830184174198718474/1137374189280571562/1151553338291269643) | [Github Pages Example](https://github.com/Pocee/PoceBlog/actions/runs/6175451494/job/16762328084#step:3:113) | [Minimal Reproduction (Stackblitz)](https://stackblitz.com/edit/github-threhe?file=src%2Fpages%2Findex.md) | [Minimal Reproduction v2](https://github.com/BryceRussell/minimal-repo-md-imgs-in-pages)
    - Breaks when using an inline relative image path in markdown located in `/src/pages`
1)  [Support Thread](https://discord.com/channels/830184174198718474/1151955934894096455/1151993152920625298) | [Example Repo](https://github.com/webpro/webpro.nl/blob/astro-issue/src/pages/og/%5B...route%5D.ts#L1)
    - Breaks when using `import.meta.glob()` to glob markdown containing inline relative image paths
1) [Support Thread](https://discord.com/channels/830184174198718474/1148668689911119922/1148668689911119922) | [Example](https://discord.com/channels/830184174198718474/1148668689911119922/1148671563286188083)
    - Breaks when importing a `.md` file with inline relative image paths
1) [Support Thread](https://discord.com/channels/830184174198718474/1147433359081996399/1147433359081996399) | [Example](https://discord.com/channels/830184174198718474/1147433359081996399/1147464631007326248)
    - Breaks when exporting result of `getImage()` in `.astro` file
    
All cases are using v3 optimized images, 5/6 of them are using inline relative image paths in markdown. 

**Example of broken output**:

![image](https://github.com/withastro/astro/assets/19967622/fa0e6bd2-8e6c-4e02-8d52-af0482d401c8)

### What's the expected result?

Build should throw an error or continue building

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-threhe?file=src%2Fpages%2Findex.md

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Pocee** [comment]
<3

**Comment by lilnasy** [comment]
Maybe a p5, not sure how widely it affects users.

**Comment by lilnasy** [comment]
Another instance encountered by @webpro.

[Thread](https://discord.com/channels/830184174198718474/1151955934894096455/1151993152920625298) | [Repo](https://github.com/webpro/webpro.nl/blob/c08c331/src/pages/og/%5B...route%5D.ts)
 - Breaks when import.meta.glob is called for content pages, unclear which markdown file is the cause.

**Comment by BryceRussell** [comment]
> Another instance encountered by @webpro.
> 
> [Thread](https://discord.com/channels/830184174198718474/1151955934894096455/1151993152920625298) | [Repo](https://github.com/webpro/webpro.nl/blob/c08c331/src/pages/og/%5B...route%5D.ts)
> 
> - Breaks when import.meta.glob is called for content pages, unclear which markdown file is the cause.

Confirmed that this is the same issue, I will add it to the list of cases above. I removed all relative image paths in the markdown from `/src/content` and the build completed successfully.

**Comment by Princesseuh** [comment]
I'll take a look, thank you for all the reproductions!

**Comment by webpro** [comment]
Just a quick note I thought I might have introduced this bug myself in #8475 (released as part of v3.1.0), but this issue is reported based on v3.0.8.
